### ISSUE TITLE
Can't case analysis on or create or[..,and[...],...] judgment

### ISSUE THREAD

**Issue by boyland** [boyland/sasylf#69]
If one of the children of an `or` judgment is an `and` judgment, the `case or` syntax doesn't put the judgments into scope.  Furthermore (a separate bug), it is not possible to prove such a judgment given the pieces of the `and` judgment.

Example:
```
theorem test:
    forall n
    exists (n = S n' and n > 0) or n = 0.
    proof by induction on n:
        case 0 is
            _: 0 = 0 by rule eq
        end case

        case S n' is
            eq: n = S n' by rule eq
            o: (n' = S n'' and n' > 0) or n' = 0 by induction hypothesis on n'
            g: n > 0 by case analysis on o:
                case or eq': (n' = (S n'')) and g': (n' > 0) is
                    _: S n' > 0 by rule gt-s on g'  // No derivation found for g'
                end case

                case or eq': n' = 0 is
                    use inversion of eq on eq'
                    proof by rule gt-1
                end case

            end case analysis
            proof by eq,g // Claimed fact is not a conjunction
        end case

    end induction
end theorem
```

**Comment by boyland** [comment]
Fixing the first part is fairly easy.  The second is harder because we currently require complete match for one disjunct.

**Comment by boyland** [comment]
The second problem can be declared a feature, since it already the case that `proof by` for an `or` judgment doesn't work: one has to give the result explicitly.  In this case, one can write
```
_: n = S n' and _: n > 0 by eq,g
```
This is accepted.

**Comment by boyland** [comment]
FIxed by commit 3e1c4a97a1bdea7028b0331cf50822c756bc46df.
